ext/standard/string.c

```diff
@@ -3863,6 +3878,55 @@ PHPAPI zend_string *php_addcslashes_str(const char *str, size_t len, const char
 }
 /* }}} */
 
+static const char *get_char_repr(char c) {
+    for (size_t i = 0; i < sizeof(char_reprs) / sizeof(char_reprs[0]); i++) {
+        if (char_reprs[i].c == c) {
+            return char_reprs[i].repr;
+        }
+    }
```

Comment:
This loop is executed for every character of the input string. It makes sense to collect the necessary symbols (similar to ``php_charmask()`` or use constant bitmask like https://github.com/php/php-src/blob/master/ext/json/json_encoder.c#L390) to make the following tests cheaper.

Reply:
Thanks for your valuable feedback! 

Can we just hardcode the char mapping in the code since it is a static mapping relation which might not need to be recalculated everytime user calls `php_repr_str`?

The hardcoded approach might bloat the code by adding 256 entries in this file.

Reply:
Solved by adding a static mapping array. 

I also conducted a simple performance (just count the CPU ticks) comparison. The following result shows the new version have **saved 32.56% CPU time**.

```
performance comparison  before        after
number of calls:        1000000       1000000
round of tests:         10            10
cpu ticks avg:          358170        241565 (-32.56%)
cpu ticks median:       356033        241496 (-32.17%)
cpu ticks stdev:        6564          372    (-94.33%)
```

Thank you so much for pointing out this performance considerations. I have learned a lot. Thanks! By the way, is there any performance benchmarking tool / framework for the Zend engine? I didn't find how to benchmark the performance so I wrote a simple c program to collect and compare the results. It would be better to have an automated benchmarking framework.

```c
zend_string *str = zend_string_init("Hello World", strlen("Hello World"), 0);
long usedTime = 0;
long numTests = 10;
int rounds = 1000000;
long *timeArray = malloc(numTests * sizeof(long));

for (int i = 0; i < numTests; i++) {
    usedTime = 0;
    for (int j = 0; j < rounds; j++) {
        long startTime = clock();
        zend_string *escaped_string = php_repr_str((str)->val, (str)->len);
        long endTime = clock();
        usedTime += endTime - startTime;
        zend_string_release(escaped_string);
    }
    timeArray[i] = usedTime;
}
```

Reply:
@dstogov Please check it out if you have time, thanks a lot!